src/darwin/Framework/CHIP/CHIPDeviceController.mm

```diff
@@ -352,6 +353,47 @@ - (BOOL)pairDevice:(uint64_t)deviceID onboardingPayload:(NSString *)onboardingPa
     return success;
 }
 
+- (BOOL)commissionDevice:(uint64_t)deviceId
+     commissioningParams:(CHIPCommissioningParameters *)commissioningParams
+                   error:(NSError * __autoreleasing *)error
+{
+    __block CHIP_ERROR errorCode = CHIP_ERROR_INCORRECT_STATE;
+    __block BOOL success = NO;
+    if (![self isRunning]) {
+        success = ![self checkForError:errorCode logMsg:kErrorNotRunning error:error];
+        return success;
+    }
+    dispatch_sync(_chipWorkQueue, ^{
+        if ([self isRunning]) {
+            chip::Controller::CommissioningParameters params;
+            if (commissioningParams.CSRNonce) {
+                params.SetCSRNonce(
+                    chip::ByteSpan((uint8_t *) commissioningParams.CSRNonce.bytes, commissioningParams.CSRNonce.length));
```

Comment:
As a followup: we have an `asByteSpan` in CHIPCluster for this.  Maybe we should move it somewhere shared and use it here.

Reply:
That would be nice. Because this got ugly.